env/fs_posix.cc

```diff
@@ -431,14 +435,16 @@ class PosixFileSystem : public FileSystem {
     }
 
     if (options.use_mmap_writes) {
-      if (!checkedDiskForMmap_) {
-        // this will be executed once in the program's lifetime.
-        // do not use mmapWrite on non ext-3/xfs/tmpfs systems.
-        if (!SupportsFastAllocate(fname)) {
-          forceMmapOff_ = true;
+      std::call_once(check_disk_for_mmap_once_, [&fname, this]() {
+        if (!checkedDiskForMmap_) {
```

Comment:
I am not sure what the point of this optimization is.  I think the code calls another fstat on to determine if range sync is supported, and there might be another one as well.  Is it possible to somehow combine these optimizations into one system call?

Reply:
`fstatfs` yes. It is tricky because of conditional compilation etc. but should be possible. (I don't recommend attempting in this PR, so that the TSAN fix is low risk.)